src/devices/Arduino/ArduinoBoard.cs

```diff
@@ -0,0 +1,421 @@
+        /// </summary>
+        /// <param name="settings">Spi Connection settings</param>
+        /// <returns>An <see cref="SpiDevice"/> instance.</returns>
+        /// <exception cref="NotSupportedException">The Bus number is not 0, or the SPI component has not been enabled in the firmware.</exception>
+        public virtual SpiDevice CreateSpiDevice(SpiConnectionSettings settings)
+        {
+            if (settings.BusId != 0)
+            {
+                throw new NotSupportedException("Only Bus Id 0 is supported");
+            }
+
+            if (!SupportedPinConfigurations.Any(x => x.PinModes.Contains(SupportedMode.SPI)))
+            {
+                throw new NotSupportedException("No Pins with SPI support found. Is the SPI module loaded?");
+            }
+
+            return new ArduinoSpiDevice(this, settings);
+        }
+
+        /// <summary>
+        /// Creates a PWM channel.
+        /// </summary>
+        /// <param name="chip">Must be 0.</param>
+        /// <param name="channel">Pin number to use</param>
+        /// <param name="frequency">This value is ignored</param>
+        /// <param name="dutyCyclePercentage">The duty cycle as a fraction.</param>
+        /// <returns></returns>
+        public virtual PwmChannel CreatePwmChannel(
+            int chip,
+            int channel,
+            int frequency = 400,
+            double dutyCyclePercentage = 0.5)
+        {
+            return new ArduinoPwmChannel(this, chip, channel, frequency, dutyCyclePercentage);
+        }
+
+        /// <summary>
+        /// Creates an anlog controller for this board.
+        /// </summary>
+        /// <param name="chip">Must be 0</param>
+        /// <returns>An <see cref="AnalogController"/> instance</returns>
+        public virtual AnalogController CreateAnalogController(int chip)
+        {
+            return new ArduinoAnalogController(this, SupportedPinConfigurations, PinNumberingScheme.Logical);
+        }
+
+        /// <summary>
+        /// Special function to read DHT sensor, if supported
+        /// </summary>
+        /// <param name="pinNumber">Pin Number</param>
+        /// <param name="dhtType">Type of DHT Sensor: 11 = DHT11, 22 = DHT22, etc.</param>
+        /// <param name="temperature">Temperature</param>
+        /// <param name="humidity">Relative humidity</param>
+        /// <returns>True on success, false otherwise</returns>
+        public bool TryReadDht(int pinNumber, int dhtType, out Temperature temperature, out RelativeHumidity humidity)
+        {
+            if (!_supportedPinConfigurations[pinNumber].PinModes.Contains(SupportedMode.DHT))
+            {
+                temperature = default;
+                humidity = default;
+                return false;
+            }
+
+            return Firmata.TryReadDht(pinNumber, dhtType, out temperature, out humidity);
+        }
```

Comment:
is DHT actually part of the spec? why is DHT special enough to be here?

Reply:
There's a proposed spec for this (I'm still waiting on getting the rights to bring the specs to the next generation). It's not in the main firmware, but the advanced firmware (see Readme) supports it. 

The reason this needs a special command (and firmware module) is that this sensor's timing is so critical, that it doesn't work when using "our" binding and a default GPIO pin. See "Limitations" in Readme.